experimental/rules_python/wheelmaker.py

```diff
@@ -157,12 +158,26 @@ def add_metadata(self, extra_headers, description, classifiers, requires,
         metadata += "\n"
         self.add_string(self.distinfo_path('METADATA'), metadata)
 
-    def add_entry_points(self, console_scripts):
+    def add_entry_points(self, console_scripts, plugins):
         """Write entry_points.txt file to the distribution."""
         # https://packaging.python.org/specifications/entry-points/
-        if not console_scripts:
+        import re
+        pattern = re.compile(r'([^=:]+):([^:=]+)=([^=]+)')
+        lines = []
+        if console_scripts:
+            lines += ["[console_scripts]"] + console_scripts
+        if plugins:
+            for line in plugins:
+                match = pattern.fullmatch(line)
+                if not match:
+                    raise ValueError('{line} is not a valid entry point'.format(line=line))
+                plugin_type = match.group(1).strip()
```

Comment:
This can produce multiple groups with the same name. Even if that's allowed by the spec (I'm not sure?) it would be better to order the entries so that each group appears only once.

Reply:
This did not happen before (at least from `wheel.bzl`) because there was only one entry allowed per group. As part of switching that to allow multiple entry points per group, rewrite this code from scratch. I believe all such concerns are now handled.